addons/mail_group/tests/test_mail_group.py

```diff
@@ -78,6 +78,51 @@ def test_find_member(self):
         member = self.test_group._find_member(email, partner_2.id)
         self.assertFalse(member, 'Should not return any member because the only one with the same email has a different partner')
 
+    def test_find_member_for_alias(self):
+        """Test the matching of a mail_group.members, when 2 users have the same partner email, and
+        that the first user was subscribed."""
+        User = self.env['res.users'].with_context(no_reset_password=True)
+        email = 'Alias <test_find_member_for_alias@example.com>'
+
+        alias = self.env['mail.alias'].create({
+            'alias_name': 'test_find_member_for_alias',
+            'alias_model_id': self.env.ref('base.model_res_partner').id,
+            'alias_contact': 'followers',
+        })
+
+        user = User.create({
+            'login': 'login_1',
+            'name': 'login_1',
+            'email': email,
+        })
+        User.create({
+            'login': 'login_2',
+            'name': 'login_2',
+            'email': email,
+        })
+
+        member = self.env['mail.group.member'].create({
+            'partner_id': user.partner_id.id,
+            'mail_group_id': self.test_group.id,
+        })
+        self.assertEqual(member.email, email)
+
+        self.test_group.access_mode == 'followers'
+        self.assertEqual(self.test_group._find_member(email, user.id), None)
+        self.assertEqual(self.test_group._find_member(email, False), member)
+
+        # Should not return string (exception) if at least one members have the same email, whatever
+        # the user_id (author_id) that could match this email.
+        #
+        # if return None, it means no error
+        # if return string (e.g. Only members can send email to the mailing list.), it means an error occurs
+        msg_dict = {
+            'author_id': self.env['mail.thread']._mail_find_user_for_gateway(email).id,
+            'email_from': email,
+        }
+        self.assertEqual(msg_dict['author_id'], user.id)
+        self.assertEqual(self.test_group._alias_get_error_message({}, msg_dict, alias), None)
```

Comment:
This seems like testing implementation details. Shouldn't you make a real email go through the mailgateway and check all final details are ok (aka: no crash, author correctly set, ...) ?

Reply:
Yes, but where ?
In CRM?... Because if I want to check if I can create an alias that create a lead... Project?... to create a task  
Base?... to create  a partner -> none sense

Purpose of this test was to test the function that I fix in this commit and so only: _alias_get_error_message

In case I create a complete alias, I call message_route and co... I will test a lot of others features and probably overlap others tests.

Reply:
Well, it seems the bug is linked to the process of posting a message on a mail.group, unless I misunderstood something.

Idea would therefore be: create a group with its alias (giving alias_name should be sufficient as normally a mail.group configures its alias correctly). Maybe update its alias_contact to followers. Then with a ``self.mock_mail_gateway()`` call ``format_and_process`` using a template (already existing in test_mail, a simple import of MAIL_TEMPLATE should do the trick ). In the end, we check that a message is posted with the right user's partner as author. Seems this is what we are fixing ?

You may have a look at tests in ``test_mail_gateway.py`` where we to a lot of that kind of tests. Globally idea is that gateway details should not be tested, only input / output (aka: right user taken into account for mail groups). Would that cover the issue ?

Reply:
> In the end, we check that a message is posted with the right user's partner as author. Seems this is what we are fixing ?

not really, this fix allow you to post a message in the group... not really change the author... Before this commit, _alias_get_error_message return that you cannot post due to the wrong way to match the allowed partner

It is why I don't reimplement a test of mail_alias complete... but just the function of matching it self